src/lib.rs

```diff
@@ -112,33 +100,54 @@ enum ProcessingError {
     },
 }
 
-impl From<WriteAheadLogError> for ProcessingError {
-    fn from(e: WriteAheadLogError) -> Self {
-        ProcessingError::General { source: e.into() }
-    }
-}
 impl From<DeltaWriterError> for ProcessingError {
     fn from(e: DeltaWriterError) -> Self {
         ProcessingError::General { source: e.into() }
     }
 }
 
-struct ProcessingState {
-    delta_writer: DeltaWriter,
-    value_buffers: ValueBuffers,
-    transformer: Transformer,
-    wal: Arc<dyn WriteAheadLog + Sync + Send>,
-    latency_timer: Instant,
+/// Application options
+pub struct Options {
+    /// Source kafka topic to consume messages from
+    pub topic: String,
+    /// Destination delta table location to produce messages into
+    pub table_location: String,
+    /// Application identifier. Must be the same among concurrent workers. Used by delta protocol
+    /// to track offsets among topic partitions.
```

Comment:
```suggestion
    /// Unique per topic per environment. Must be the same for all processes that are part of a single job.
```

@houqp Okay, so the _worker_ was obsolete as I intended to use new option _worker_name_  for debugging purpose but found a way around it.

So here's the suggestion to use the same as in https://github.com/delta-io/kafka-delta-ingest/blob/main/doc/DESIGN.md#process-parameters and remove mention of worker, wdyt?

cc @xianwill

Reply:
got it, unique per topic per environment makes sense to me. what does a single job refer to here? looks like a single job can be consists of multiple ecs tasks?

Reply:
@xianwill Already answered this https://github.com/delta-io/kafka-delta-ingest/blob/main/doc/DESIGN.md#architecture, nice :)

Reply:
The job reference might take place in the docs for KafkaJsonToDelta or something. I feel like we should introduce a convention to add docs on each newly created struct / pub func. And then wrap it up with some PR like in delta-rs to cover everything else once we got time